pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -46,6 +49,7 @@
 public class DataSchema {
   private final String[] _columnNames;
   private final ColumnDataType[] _columnDataTypes;
+  private final FieldSpec[] _columnFieldSpecs;
```

Comment:
Trying to understand why we need `FieldSpec` in the data schema. Seems the intention is to use it to calculate the default null value.
It should not be stored here for the following reasons:
1. We don't want to carry the field spec in the transport object
2. The column could be an expression (transform function/aggregation function etc), and there is no existing `FieldSpec` for them
3. We should not need to calculate default null value on the fly. Filling any value should be fine because we should skip processing them